File: src/main/java/org/jukeboxmc/world/manager/PopulationChunkManager.java
package org.jukeboxmc.world.manager;

import lombok.RequiredArgsConstructor;
import org.jukeboxmc.world.chunk.Chunk;

@RequiredArgsConstructor
public final class PopulationChunkManager implements IChunkManager {

    private final int centerX;
    private final int centerZ;
    private final Chunk[] chunks;

    @Override
    public Chunk getChunk( int x, int z ) {
        return this.chunks[( x - this.centerX + 1 ) + ( z - this.centerZ + 1 ) * 3];
    }

}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/world/generator/PopulationChunkManager.kt
package org.jukeboxmc.api.world.generator

import org.jukeboxmc.api.block.Block
import org.jukeboxmc.api.math.Vector
import org.jukeboxmc.api.world.chunk.Chunk

interface PopulationChunkManager {

    fun getChunk(x: Int, z: Int): Chunk

    fun getBlock(x: Int, y: Int, z: Int): Block

    fun getBlock(x: Int, y: Int, z: Int, layer: Int): Block

    fun getBlock(location: Vector, layer: Int): Block

    fun getBlock(location: Vector): Block

    fun setBlock(x: Int, y: Int, z: Int, layer: Int, block: Block)

    fun setBlock(x: Int, y: Int, z: Int, block: Block)

    fun setBlock(location: Vector, layer: Int, block: Block)

    fun setBlock(location: Vector, block: Block)
}